src/HttpClientRequestAdapter.cs

```diff
@@ -293,7 +295,23 @@ public ISerializationWriterFactory SerializationWriterFactory
                         {
                             result = rootNode.GetDateValue();
                         }
-                        else throw new InvalidOperationException("error handling the response, unexpected type");
+                        else if(
+                            Nullable.GetUnderlyingType(modelType) is { IsEnum: true } underlyingType &&
+                            rootNode.GetStringValue() is { Length: > 0 } rawValue)
+                        {
+                            foreach(var field in underlyingType.GetFields())
+                            {
+                                if(field.GetCustomAttribute<EnumMemberAttribute>() is { } attr && rawValue.Equals(attr.Value, StringComparison.Ordinal))
+                                {
+                                    rawValue = field.Name;
+                                }

```

Comment:
isn't enum.Parse already using the member attributes?
Also, what do you think about breaking the loop whenever the value is found so we save a few iterations?

Reply:
enum.Parse will throw ArgumentException if you try to parse value defined in EnumMember attribute.
Yes, break was missing in the loop